### PR TITLE
andreasbuhr-cppcoro: migrate to Conan v2

### PR DESCRIPTION
**Pull request by valgur**

### CODE REVIEW

**Comment by valgur** [comment]
Closing temporarily to avoid unnecessary load on the CI. Will reopen when I'm actively working on the PR again.

File: recipes/andreasbuhr-cppcoro/all/conanfile.py [hunk 1]
```diff
@@ -45,68 +49,73 @@ def config_options(self):
...
-        # This should be removed after both gcc and clang implements the final coroutine TS
+        # This should be removed after both gcc and clang implement the final coroutine TS
         if self.settings.compiler == "clang" and self.settings.compiler.get_safe("libcxx") == "libstdc++":
-            raise ConanInvalidConfiguration("{} does not support clang with libstdc++. Use libc++ instead.".format(self.name))
-
-    def configure(self):
-        if self.options.shared:
-            del self.options.fPIC
+            raise ConanInvalidConfiguration(
+                f"{self.name} does not support clang with libstdc++. Use libc++ instead."
```
Comment (RubenRBS) [inline recipes/andreasbuhr-cppcoro/all/conanfile.py:76]: Is this still the case? https://godbolt.org/z/GndEK54GY compiles for me

File: recipes/andreasbuhr-cppcoro/all/conanfile.py [hunk 2]
```diff
@@ -45,68 +49,73 @@ def config_options(self):
...
+            self.cpp_info.cxxflags.append("/await")
         elif self.settings.compiler == "gcc":
-            comp.cxxflags.append("-fcoroutines")
-            comp.defines.append("CPPCORO_COMPILER_SUPPORTS_SYMMETRIC_TRANSFER=1")
-        elif self.settings.compiler == "clang" or self.settings.compiler == "apple-clang":
-            comp.cxxflags.append("-fcoroutines-ts")
+            self.cpp_info.cxxflags.append("-fcoroutines")
+            self.cpp_info.defines.append("CPPCORO_COMPILER_SUPPORTS_SYMMETRIC_TRANSFER=1")
+        elif self.settings.compiler in ["clang", "apple-clang"]:
+            self.cpp_info.cxxflags.append("-fcoroutines-ts")
```
Comment (RubenRBS) [inline recipes/andreasbuhr-cppcoro/all/conanfile.py:115]: This flag is no longer valid in newer clang versions, should check its version beforhand

**Comment by valgur** [comment]
Closing in favor of #20611.

**Pull request closed without merging**